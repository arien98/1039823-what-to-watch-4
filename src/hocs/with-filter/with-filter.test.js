import React from "react";
import renderer from "react-test-renderer";
import withFilter from "./with-filter";

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};

const MockComponentWrapped = withFilter(MockComponent);

describe(`with filter snapshot`, () => {
  test(`it renders correctly`, () => {
    const props = {
      filmsData: [
        {
          id: 2,
          title: `Bohemian Rhapsody`,
          poster: `img/bohemian-rhapsody.jpg`,
        },
        {
          id: 3,
          title: `Macbeth`,
          poster: `img/macbeth.jpg`,
        },
        {
          id: 4,
          title: `Aviator`,
          poster: `img/aviator.jpg`,
        },
      ],
      onFilmCardClick: () => {},
    };

    const tree = renderer
      .create(<MockComponentWrapped {...props}></MockComponentWrapped>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
