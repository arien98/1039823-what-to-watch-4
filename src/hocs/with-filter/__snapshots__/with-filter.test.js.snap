// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with filter snapshot it renders correctly 1`] = `<div />`;
